src/pg_local.erl

```diff
@@ -148,46 +157,57 @@ terminate(_Reason, _S) ->
 %%% {{pid, Pid, Name}}
 %%%    Pid is a member of group Name.
 
-member_died(Ref) ->
-    [{{ref, Ref}, Pid}] = ets:lookup(pg_local_table, {ref, Ref}),
+member_died(Ref, Pid) ->
+    case ets:lookup(?TABLE, {ref, Ref}) of
```

Comment:
Do we really need this lookup if we always call `leave_all_groups` with the same `Pid`? The three `case` clauses do exactly the same. Or am I missing something?

Reply:
This code is many years old so I can only guess the intent here. I think it tries to assert that the ref/pid combination is known. I'd leave it as is for now, this is not a hot code path event.

Reply:
The clauses simply demonstrate what 3 scenarios are possible. We can combine the latter two.